app/main/authorize/access_token_sign_in_required.py

```diff
@@ -63,18 +79,30 @@ def decorated_view(*args, **kwargs):
     return decorated_view
 
 
-def _validate_access_token():
-    is_valid_access_token = False
+def _validate_or_refresh_tokens(access_token, refresh_token):
+    tokens_are_refreshed = False
+
+    if not (access_token and refresh_token):
```

Comment:
I made the assumption that we should always have access_token and refresh_token set due to us setting them both in the callback and nobody can manipulate the session due to the flask secret.

This assumption keeps the test cases simpler

Reply:
good and precise clarification, thank you